core/txpool/txpool.go

```diff
@@ -706,17 +734,34 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (replaced bool, err e
 			overflowedTxMeter.Mark(1)
 			return false, ErrTxPoolOverflow
 		}
+
+		// If the new transaction is a future transaction it should never churn pending transactions
+		if pool.isFuture(from, tx) {
+			for i := 0; i < len(drop); i++ {
+				dropSender, _ := types.Sender(pool.signer, drop[i])
+				if list := pool.pending[dropSender]; list != nil && list.Overlaps(drop[i]) {
+					// Add all transactions back to the priced queue
+					for k := 0; k < len(drop); k++ {
+						heap.Push(&pool.priced.urgent, drop[k])
+					}
+					log.Trace("Discarding future transaction replacing pending tx", "hash", hash)
+					return false, ErrFutureReplacePending
+				}
+			}
+		}
+
 		// Bump the counter of rejections-since-reorg
 		pool.changesSinceReorg += len(drop)
 		// Kick out the underpriced remote transactions.
 		for _, tx := range drop {
 			log.Trace("Discarding freshly underpriced transaction", "hash", tx.Hash(), "gasTipCap", tx.GasTipCap(), "gasFeeCap", tx.GasFeeCap())
 			underpricedTxMeter.Mark(1)
-			pool.removeTx(tx.Hash(), false)
+			dropped := pool.removeTx(tx.Hash(), false)
+			pool.changesSinceReorg += dropped
```

Comment:
This is already counted at line 754

Reply:
I guess it was my bad -- previously, you had it correct. The upper one counted the direct drops, the `dropped` here counted the number of indirect movements from pending to queued.

Reply:
If there's anything That you need to figure it out please. Contact me.